src/pages/api/modules/usenet/index.ts

```diff
@@ -0,0 +1,70 @@
+import { getCookie } from 'cookies-next';
+import dayjs from 'dayjs';
+import duration from 'dayjs/plugin/duration';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { Client } from 'sabnzbd-api';
+import { getConfig } from '../../../../tools/getConfig';
+import { getServiceById } from '../../../../tools/hooks/useGetServiceByType';
+import { Config } from '../../../../tools/types';
+
+dayjs.extend(duration);
+
+export interface UsenetInfoRequestParams {
+  serviceId: string;
+}
+
+export interface UsenetInfoResponse {
+  paused: boolean;
+  sizeLeft: number;
+  speed: number;
+  eta: number;
+}
+
+async function Get(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    const configName = getCookie('config-name', { req });
+    const { config }: { config: Config } = getConfig(configName?.toString() ?? 'default').props;
+    const { serviceId } = req.query as any as UsenetInfoRequestParams;
+
+    const service = getServiceById(config, serviceId);
+
+    if (!service) {
+      throw new Error(`Service with ID "${req.query.serviceId}" could not be found.`);
+    }
+
+    if (!service.apiKey) {
+      throw new Error(`API Key for service "${service.name}" is missing`);
+    }
+
+    const queue = await new Client(service.url, service.apiKey).queue(0, -1);
+
+    const [hours, minutes, seconds] = queue.timeleft.split(':');
+    const eta = dayjs.duration({
+      hour: parseInt(hours, 10),
+      minutes: parseInt(minutes, 10),
+      seconds: parseInt(seconds, 10),
+    } as any);
```

Comment:
Why use so many anys, isn't dayjs typed ?

Reply:
Yeah but the types aren't quite right. They require everything here but it should only be a `Partial`